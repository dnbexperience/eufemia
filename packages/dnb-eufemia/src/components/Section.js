/**
 * ATTENTION: This file is auto generated by using "prepareTemplates".
 * Do not change the content!
 *
 */

/**
 * Library Index section to autogenerate all the components and patterns
 * Used by "prepareSections"
 */

import Section from './section/Section'
export * from './section/Section'
export default Section
