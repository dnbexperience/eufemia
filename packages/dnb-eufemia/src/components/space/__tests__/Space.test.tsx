/**
 * Space Test
 *
 */

import React from 'react'
import { loadScss } from '../../../core/jest/jestSetup'
import { render } from '@testing-library/react'
import Space, { SpaceAllProps } from '../Space'
import Section from '../../Section'

const props: SpaceAllProps = {}

describe('Space component', () => {
  it('renders with empty props', () => {
    render(<Space {...props} />)
    expect(document.querySelector('.dnb-space')).toBeInTheDocument()
  })

  it('should have correct CSS classes', () => {
    render(<Space element="span" top="large" />)
    expect(document.querySelector('span.dnb-space').classList).toContain(
      'dnb-space__top--large'
    )
  })

  it('gets valid ref element', () => {
    let ref: React.RefObject<HTMLDivElement>

    function MockComponent(props) {
      ref = React.useRef()
      return <Space {...props} innerRef={ref} />
    }

    const { rerender } = render(<MockComponent />)

    expect(ref.current instanceof HTMLDivElement).toBe(true)
    expect(ref.current.tagName).toBe('DIV')

    rerender(<MockComponent element={Section} />)

    expect(ref.current instanceof HTMLElement).toBe(true)
    expect(ref.current.tagName).toBe('SECTION')
  })

  it('should accept space only prop', () => {
    render(<Space element="span" space="large" />)
    expect(
      Object.values(document.querySelector('span.dnb-space').classList)
    ).toEqual([
      'dnb-space',
      'dnb-space__top--large',
      'dnb-space__right--large',
      'dnb-space__bottom--large',
      'dnb-space__left--large',
    ])
  })

  it('should accept space prop as an object with spacing properties', () => {
    render(
      <Space
        element="span"
        space={{
          top: 'x-large',
          right: 'large',
          bottom: 'small small',
          left: 'x-small x-small small',
        }}
      />
    )
    expect(
      Object.values(document.querySelector('span.dnb-space').classList)
    ).toEqual([
      'dnb-space',
      'dnb-space__top--x-large',
      'dnb-space__right--large',
      'dnb-space__bottom--large',
      'dnb-space__left--large',
    ])
  })

  it('should accept id attribute', () => {
    render(<Space id="custom-id" />)

    const element = document.querySelector('div.dnb-space')
    const attributes = Array.from(element.attributes).map(
      (attr) => attr.name
    )

    expect(attributes).toEqual(['class', 'id'])
    expect(element.getAttribute('id')).toBe('custom-id')
  })

  it('should have collapse CSS class', () => {
    render(<Space top="large" no_collapse={true} />)
    expect(
      document.querySelector('.dnb-space--no-collapse')
    ).toBeInTheDocument()
  })

  it('should support "innerSpace"', () => {
    const { rerender } = render(<Space innerSpace={true} />)
    expect(
      document.querySelector('.dnb-space').getAttribute('style')
    ).toBe(
      '--space-t-s: 1rem; --space-r-s: 1rem; --space-b-s: 1rem; --space-l-s: 1rem; --space-t-m: 1rem; --space-r-m: 1rem; --space-b-m: 1rem; --space-l-m: 1rem; --space-t-l: 1rem; --space-r-l: 1rem; --space-b-l: 1rem; --space-l-l: 1rem;'
    )

    rerender(<Space innerSpace="large medium small" />)
    expect(
      document.querySelector('.dnb-space').getAttribute('style')
    ).toBe(
      '--space-t-s: 4.5rem; --space-r-s: 4.5rem; --space-b-s: 4.5rem; --space-l-s: 4.5rem; --space-t-m: 4.5rem; --space-r-m: 4.5rem; --space-b-m: 4.5rem; --space-l-m: 4.5rem; --space-t-l: 4.5rem; --space-r-l: 4.5rem; --space-b-l: 4.5rem; --space-l-l: 4.5rem;'
    )

    rerender(
      <Space
        innerSpace={{
          small: { top: '0.5rem', right: 'large' },
          medium: true,
          large: { left: '16px', right: 'x-small' },
        }}
      />
    )
    expect(
      document.querySelector('.dnb-space').getAttribute('style')
    ).toBe(
      '--space-t-s: 0.5rem; --space-r-s: 2rem; --space-t-m: 1rem; --space-r-m: 1rem; --space-b-m: 1rem; --space-l-m: 1rem; --space-r-l: 0.5rem; --space-l-l: 1rem;'
    )
  })
})

describe('Space scss', () => {
  it('has to match style dependencies css', () => {
    const css = loadScss(require.resolve('../style/deps.scss'))
    expect(css).toMatchSnapshot()
  })

  it('have to match default theme snapshot', () => {
    const css = loadScss(
      require.resolve('../style/themes/dnb-space-theme-ui.scss')
    )
    expect(css).toMatchSnapshot()
  })
})
