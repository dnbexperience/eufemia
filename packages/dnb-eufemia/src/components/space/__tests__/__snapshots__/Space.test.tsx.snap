// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Space scss has to match style dependencies css 1`] = `
"/*
* Used for snapshot testing
*
*/
/*
 * Utilities
 */
/*
 * Space component
 *
 * TODO: Set spacing in the theme file, as theme comes later in the stack
 *
 */
/*
 * Utilities
 */
/** NB: Because CSS specificity ".dnb-core-style .dnb-space"
    has to come before the next ".dnb-space"
  */
.dnb-core-style .dnb-space__top--zero {
  margin-top: 0;
}
.dnb-core-style .dnb-space__top--xx-small {
  margin-top: var(--spacing-xx-small);
}
.dnb-core-style .dnb-space__top--x-small {
  margin-top: var(--spacing-x-small);
}
.dnb-core-style .dnb-space__top--small {
  margin-top: var(--spacing-small);
}
.dnb-core-style .dnb-space__top--medium {
  margin-top: var(--spacing-medium);
}
.dnb-core-style .dnb-space__top--large {
  margin-top: var(--spacing-large);
}
.dnb-core-style .dnb-space__top--large.dnb-core-style .dnb-space__top--x-small {
  margin-top: calc(var(--spacing-large) + var(--spacing-x-small));
}
.dnb-core-style .dnb-space__top--x-large {
  margin-top: var(--spacing-x-large);
}
.dnb-core-style .dnb-space__top--xx-large {
  margin-top: var(--spacing-xx-large);
}
.dnb-core-style .dnb-space__top--xx-large.dnb-core-style .dnb-space__top--x-small {
  margin-top: calc(var(--spacing-large) + var(--spacing-large));
}
.dnb-core-style .dnb-space__top--xx-large.dnb-core-style .dnb-space__top--small {
  margin-top: calc(var(--spacing-x-large) + var(--spacing-medium));
}
.dnb-core-style .dnb-space__top--xx-large.dnb-core-style .dnb-space__top--medium {
  margin-top: calc(var(--spacing-x-large) + var(--spacing-large));
}
.dnb-core-style .dnb-space__top--xx-large.dnb-core-style .dnb-space__top--large {
  margin-top: calc(var(--spacing-x-large) + var(--spacing-large) + var(--spacing-x-small));
}
.dnb-core-style .dnb-space__top--xx-large.dnb-core-style .dnb-space__top--large.dnb-core-style .dnb-space__top--x-small {
  margin-top: calc(var(--spacing-x-large) + var(--spacing-large) + var(--spacing-small));
}
.dnb-core-style .dnb-space__top--xx-large.dnb-core-style .dnb-space__top--x-large {
  margin-top: calc(var(--spacing-xx-large) + var(--spacing-x-large));
}
.dnb-core-style .dnb-space__top--xx-large-x2 {
  margin-top: calc(var(--spacing-x-large) + var(--spacing-large) + var(--spacing-large));
}
.dnb-core-style .dnb-space__top--xx-large-x2.dnb-core-style .dnb-space__top--x-small {
  margin-top: calc(var(--spacing-xx-large) + var(--spacing-large) + var(--spacing-large));
}
.dnb-core-style .dnb-space__top--xx-large-x2.dnb-core-style .dnb-space__top--small {
  margin-top: calc(var(--spacing-x-large) + var(--spacing-x-large) + var(--spacing-large));
}
.dnb-core-style .dnb-space__top--xx-large-x2.dnb-core-style .dnb-space__top--medium {
  margin-top: calc(var(--spacing-xx-large) + var(--spacing-x-large) + var(--spacing-large));
}
.dnb-core-style .dnb-space__top--xx-large-x2.dnb-core-style .dnb-space__top--large {
  margin-top: calc(var(--spacing-x-large) + var(--spacing-x-large) + var(--spacing-x-large));
}
.dnb-core-style .dnb-space__top--xx-large-x2.dnb-core-style .dnb-space__top--large.dnb-core-style .dnb-space__top--x-small {
  margin-top: calc(var(--spacing-xx-large) + var(--spacing-x-large) + var(--spacing-x-large));
}
.dnb-core-style .dnb-space__top--xx-large-x2.dnb-core-style .dnb-space__top--x-large {
  margin-top: calc(var(--spacing-xx-large) + var(--spacing-xx-large) + var(--spacing-x-large));
}
.dnb-core-style .dnb-space__right--zero {
  margin-right: 0;
}
.dnb-core-style .dnb-space__right--xx-small {
  margin-right: var(--spacing-xx-small);
}
.dnb-core-style .dnb-space__right--x-small {
  margin-right: var(--spacing-x-small);
}
.dnb-core-style .dnb-space__right--small {
  margin-right: var(--spacing-small);
}
.dnb-core-style .dnb-space__right--medium {
  margin-right: var(--spacing-medium);
}
.dnb-core-style .dnb-space__right--large {
  margin-right: var(--spacing-large);
}
.dnb-core-style .dnb-space__right--large.dnb-core-style .dnb-space__right--x-small {
  margin-right: calc(var(--spacing-large) + var(--spacing-x-small));
}
.dnb-core-style .dnb-space__right--x-large {
  margin-right: var(--spacing-x-large);
}
.dnb-core-style .dnb-space__right--xx-large {
  margin-right: var(--spacing-xx-large);
}
.dnb-core-style .dnb-space__right--xx-large.dnb-core-style .dnb-space__right--x-small {
  margin-right: calc(var(--spacing-large) + var(--spacing-large));
}
.dnb-core-style .dnb-space__right--xx-large.dnb-core-style .dnb-space__right--small {
  margin-right: calc(var(--spacing-x-large) + var(--spacing-medium));
}
.dnb-core-style .dnb-space__right--xx-large.dnb-core-style .dnb-space__right--medium {
  margin-right: calc(var(--spacing-x-large) + var(--spacing-large));
}
.dnb-core-style .dnb-space__right--xx-large.dnb-core-style .dnb-space__right--large {
  margin-right: calc(var(--spacing-x-large) + var(--spacing-large) + var(--spacing-x-small));
}
.dnb-core-style .dnb-space__right--xx-large.dnb-core-style .dnb-space__right--large.dnb-core-style .dnb-space__right--x-small {
  margin-right: calc(var(--spacing-x-large) + var(--spacing-large) + var(--spacing-small));
}
.dnb-core-style .dnb-space__right--xx-large.dnb-core-style .dnb-space__right--x-large {
  margin-right: calc(var(--spacing-xx-large) + var(--spacing-x-large));
}
.dnb-core-style .dnb-space__right--xx-large-x2 {
  margin-right: calc(var(--spacing-x-large) + var(--spacing-large) + var(--spacing-large));
}
.dnb-core-style .dnb-space__right--xx-large-x2.dnb-core-style .dnb-space__right--x-small {
  margin-right: calc(var(--spacing-xx-large) + var(--spacing-large) + var(--spacing-large));
}
.dnb-core-style .dnb-space__right--xx-large-x2.dnb-core-style .dnb-space__right--small {
  margin-right: calc(var(--spacing-x-large) + var(--spacing-x-large) + var(--spacing-large));
}
.dnb-core-style .dnb-space__right--xx-large-x2.dnb-core-style .dnb-space__right--medium {
  margin-right: calc(var(--spacing-xx-large) + var(--spacing-x-large) + var(--spacing-large));
}
.dnb-core-style .dnb-space__right--xx-large-x2.dnb-core-style .dnb-space__right--large {
  margin-right: calc(var(--spacing-x-large) + var(--spacing-x-large) + var(--spacing-x-large));
}
.dnb-core-style .dnb-space__right--xx-large-x2.dnb-core-style .dnb-space__right--large.dnb-core-style .dnb-space__right--x-small {
  margin-right: calc(var(--spacing-xx-large) + var(--spacing-x-large) + var(--spacing-x-large));
}
.dnb-core-style .dnb-space__right--xx-large-x2.dnb-core-style .dnb-space__right--x-large {
  margin-right: calc(var(--spacing-xx-large) + var(--spacing-xx-large) + var(--spacing-x-large));
}
.dnb-core-style .dnb-space__bottom--zero {
  margin-bottom: 0;
}
.dnb-core-style .dnb-space__bottom--xx-small {
  margin-bottom: var(--spacing-xx-small);
}
.dnb-core-style .dnb-space__bottom--x-small {
  margin-bottom: var(--spacing-x-small);
}
.dnb-core-style .dnb-space__bottom--small {
  margin-bottom: var(--spacing-small);
}
.dnb-core-style .dnb-space__bottom--medium {
  margin-bottom: var(--spacing-medium);
}
.dnb-core-style .dnb-space__bottom--large {
  margin-bottom: var(--spacing-large);
}
.dnb-core-style .dnb-space__bottom--large.dnb-core-style .dnb-space__bottom--x-small {
  margin-bottom: calc(var(--spacing-large) + var(--spacing-x-small));
}
.dnb-core-style .dnb-space__bottom--x-large {
  margin-bottom: var(--spacing-x-large);
}
.dnb-core-style .dnb-space__bottom--xx-large {
  margin-bottom: var(--spacing-xx-large);
}
.dnb-core-style .dnb-space__bottom--xx-large.dnb-core-style .dnb-space__bottom--x-small {
  margin-bottom: calc(var(--spacing-large) + var(--spacing-large));
}
.dnb-core-style .dnb-space__bottom--xx-large.dnb-core-style .dnb-space__bottom--small {
  margin-bottom: calc(var(--spacing-x-large) + var(--spacing-medium));
}
.dnb-core-style .dnb-space__bottom--xx-large.dnb-core-style .dnb-space__bottom--medium {
  margin-bottom: calc(var(--spacing-x-large) + var(--spacing-large));
}
.dnb-core-style .dnb-space__bottom--xx-large.dnb-core-style .dnb-space__bottom--large {
  margin-bottom: calc(var(--spacing-x-large) + var(--spacing-large) + var(--spacing-x-small));
}
.dnb-core-style .dnb-space__bottom--xx-large.dnb-core-style .dnb-space__bottom--large.dnb-core-style .dnb-space__bottom--x-small {
  margin-bottom: calc(var(--spacing-x-large) + var(--spacing-large) + var(--spacing-small));
}
.dnb-core-style .dnb-space__bottom--xx-large.dnb-core-style .dnb-space__bottom--x-large {
  margin-bottom: calc(var(--spacing-xx-large) + var(--spacing-x-large));
}
.dnb-core-style .dnb-space__bottom--xx-large-x2 {
  margin-bottom: calc(var(--spacing-x-large) + var(--spacing-large) + var(--spacing-large));
}
.dnb-core-style .dnb-space__bottom--xx-large-x2.dnb-core-style .dnb-space__bottom--x-small {
  margin-bottom: calc(var(--spacing-xx-large) + var(--spacing-large) + var(--spacing-large));
}
.dnb-core-style .dnb-space__bottom--xx-large-x2.dnb-core-style .dnb-space__bottom--small {
  margin-bottom: calc(var(--spacing-x-large) + var(--spacing-x-large) + var(--spacing-large));
}
.dnb-core-style .dnb-space__bottom--xx-large-x2.dnb-core-style .dnb-space__bottom--medium {
  margin-bottom: calc(var(--spacing-xx-large) + var(--spacing-x-large) + var(--spacing-large));
}
.dnb-core-style .dnb-space__bottom--xx-large-x2.dnb-core-style .dnb-space__bottom--large {
  margin-bottom: calc(var(--spacing-x-large) + var(--spacing-x-large) + var(--spacing-x-large));
}
.dnb-core-style .dnb-space__bottom--xx-large-x2.dnb-core-style .dnb-space__bottom--large.dnb-core-style .dnb-space__bottom--x-small {
  margin-bottom: calc(var(--spacing-xx-large) + var(--spacing-x-large) + var(--spacing-x-large));
}
.dnb-core-style .dnb-space__bottom--xx-large-x2.dnb-core-style .dnb-space__bottom--x-large {
  margin-bottom: calc(var(--spacing-xx-large) + var(--spacing-xx-large) + var(--spacing-x-large));
}
.dnb-core-style .dnb-space__left--zero {
  margin-left: 0;
}
.dnb-core-style .dnb-space__left--xx-small {
  margin-left: var(--spacing-xx-small);
}
.dnb-core-style .dnb-space__left--x-small {
  margin-left: var(--spacing-x-small);
}
.dnb-core-style .dnb-space__left--small {
  margin-left: var(--spacing-small);
}
.dnb-core-style .dnb-space__left--medium {
  margin-left: var(--spacing-medium);
}
.dnb-core-style .dnb-space__left--large {
  margin-left: var(--spacing-large);
}
.dnb-core-style .dnb-space__left--large.dnb-core-style .dnb-space__left--x-small {
  margin-left: calc(var(--spacing-large) + var(--spacing-x-small));
}
.dnb-core-style .dnb-space__left--x-large {
  margin-left: var(--spacing-x-large);
}
.dnb-core-style .dnb-space__left--xx-large {
  margin-left: var(--spacing-xx-large);
}
.dnb-core-style .dnb-space__left--xx-large.dnb-core-style .dnb-space__left--x-small {
  margin-left: calc(var(--spacing-large) + var(--spacing-large));
}
.dnb-core-style .dnb-space__left--xx-large.dnb-core-style .dnb-space__left--small {
  margin-left: calc(var(--spacing-x-large) + var(--spacing-medium));
}
.dnb-core-style .dnb-space__left--xx-large.dnb-core-style .dnb-space__left--medium {
  margin-left: calc(var(--spacing-x-large) + var(--spacing-large));
}
.dnb-core-style .dnb-space__left--xx-large.dnb-core-style .dnb-space__left--large {
  margin-left: calc(var(--spacing-x-large) + var(--spacing-large) + var(--spacing-x-small));
}
.dnb-core-style .dnb-space__left--xx-large.dnb-core-style .dnb-space__left--large.dnb-core-style .dnb-space__left--x-small {
  margin-left: calc(var(--spacing-x-large) + var(--spacing-large) + var(--spacing-small));
}
.dnb-core-style .dnb-space__left--xx-large.dnb-core-style .dnb-space__left--x-large {
  margin-left: calc(var(--spacing-xx-large) + var(--spacing-x-large));
}
.dnb-core-style .dnb-space__left--xx-large-x2 {
  margin-left: calc(var(--spacing-x-large) + var(--spacing-large) + var(--spacing-large));
}
.dnb-core-style .dnb-space__left--xx-large-x2.dnb-core-style .dnb-space__left--x-small {
  margin-left: calc(var(--spacing-xx-large) + var(--spacing-large) + var(--spacing-large));
}
.dnb-core-style .dnb-space__left--xx-large-x2.dnb-core-style .dnb-space__left--small {
  margin-left: calc(var(--spacing-x-large) + var(--spacing-x-large) + var(--spacing-large));
}
.dnb-core-style .dnb-space__left--xx-large-x2.dnb-core-style .dnb-space__left--medium {
  margin-left: calc(var(--spacing-xx-large) + var(--spacing-x-large) + var(--spacing-large));
}
.dnb-core-style .dnb-space__left--xx-large-x2.dnb-core-style .dnb-space__left--large {
  margin-left: calc(var(--spacing-x-large) + var(--spacing-x-large) + var(--spacing-x-large));
}
.dnb-core-style .dnb-space__left--xx-large-x2.dnb-core-style .dnb-space__left--large.dnb-core-style .dnb-space__left--x-small {
  margin-left: calc(var(--spacing-xx-large) + var(--spacing-x-large) + var(--spacing-x-large));
}
.dnb-core-style .dnb-space__left--xx-large-x2.dnb-core-style .dnb-space__left--x-large {
  margin-left: calc(var(--spacing-xx-large) + var(--spacing-xx-large) + var(--spacing-x-large));
}

/* stylelint-disable */
.dnb-space--no-collapse {
  display: flow-root;
}
.dnb-space--stretch {
  width: 100%;
}
.dnb-space--inline {
  display: inline-block;
}
.dnb-space__top--zero {
  margin-top: 0;
}
.dnb-space__top--xx-small {
  margin-top: var(--spacing-xx-small);
}
.dnb-space__top--x-small {
  margin-top: var(--spacing-x-small);
}
.dnb-space__top--small {
  margin-top: var(--spacing-small);
}
.dnb-space__top--medium {
  margin-top: var(--spacing-medium);
}
.dnb-space__top--large {
  margin-top: var(--spacing-large);
}
.dnb-space__top--large.dnb-space__top--x-small {
  margin-top: calc(var(--spacing-large) + var(--spacing-x-small));
}
.dnb-space__top--x-large {
  margin-top: var(--spacing-x-large);
}
.dnb-space__top--xx-large {
  margin-top: var(--spacing-xx-large);
}
.dnb-space__top--xx-large.dnb-space__top--x-small {
  margin-top: calc(var(--spacing-large) + var(--spacing-large));
}
.dnb-space__top--xx-large.dnb-space__top--small {
  margin-top: calc(var(--spacing-x-large) + var(--spacing-medium));
}
.dnb-space__top--xx-large.dnb-space__top--medium {
  margin-top: calc(var(--spacing-x-large) + var(--spacing-large));
}
.dnb-space__top--xx-large.dnb-space__top--large {
  margin-top: calc(var(--spacing-x-large) + var(--spacing-large) + var(--spacing-x-small));
}
.dnb-space__top--xx-large.dnb-space__top--large.dnb-space__top--x-small {
  margin-top: calc(var(--spacing-x-large) + var(--spacing-large) + var(--spacing-small));
}
.dnb-space__top--xx-large.dnb-space__top--x-large {
  margin-top: calc(var(--spacing-xx-large) + var(--spacing-x-large));
}
.dnb-space__top--xx-large-x2 {
  margin-top: calc(var(--spacing-x-large) + var(--spacing-large) + var(--spacing-large));
}
.dnb-space__top--xx-large-x2.dnb-space__top--x-small {
  margin-top: calc(var(--spacing-xx-large) + var(--spacing-large) + var(--spacing-large));
}
.dnb-space__top--xx-large-x2.dnb-space__top--small {
  margin-top: calc(var(--spacing-x-large) + var(--spacing-x-large) + var(--spacing-large));
}
.dnb-space__top--xx-large-x2.dnb-space__top--medium {
  margin-top: calc(var(--spacing-xx-large) + var(--spacing-x-large) + var(--spacing-large));
}
.dnb-space__top--xx-large-x2.dnb-space__top--large {
  margin-top: calc(var(--spacing-x-large) + var(--spacing-x-large) + var(--spacing-x-large));
}
.dnb-space__top--xx-large-x2.dnb-space__top--large.dnb-space__top--x-small {
  margin-top: calc(var(--spacing-xx-large) + var(--spacing-x-large) + var(--spacing-x-large));
}
.dnb-space__top--xx-large-x2.dnb-space__top--x-large {
  margin-top: calc(var(--spacing-xx-large) + var(--spacing-xx-large) + var(--spacing-x-large));
}
.dnb-space__right--zero {
  margin-right: 0;
}
.dnb-space__right--xx-small {
  margin-right: var(--spacing-xx-small);
}
.dnb-space__right--x-small {
  margin-right: var(--spacing-x-small);
}
.dnb-space__right--small {
  margin-right: var(--spacing-small);
}
.dnb-space__right--medium {
  margin-right: var(--spacing-medium);
}
.dnb-space__right--large {
  margin-right: var(--spacing-large);
}
.dnb-space__right--large.dnb-space__right--x-small {
  margin-right: calc(var(--spacing-large) + var(--spacing-x-small));
}
.dnb-space__right--x-large {
  margin-right: var(--spacing-x-large);
}
.dnb-space__right--xx-large {
  margin-right: var(--spacing-xx-large);
}
.dnb-space__right--xx-large.dnb-space__right--x-small {
  margin-right: calc(var(--spacing-large) + var(--spacing-large));
}
.dnb-space__right--xx-large.dnb-space__right--small {
  margin-right: calc(var(--spacing-x-large) + var(--spacing-medium));
}
.dnb-space__right--xx-large.dnb-space__right--medium {
  margin-right: calc(var(--spacing-x-large) + var(--spacing-large));
}
.dnb-space__right--xx-large.dnb-space__right--large {
  margin-right: calc(var(--spacing-x-large) + var(--spacing-large) + var(--spacing-x-small));
}
.dnb-space__right--xx-large.dnb-space__right--large.dnb-space__right--x-small {
  margin-right: calc(var(--spacing-x-large) + var(--spacing-large) + var(--spacing-small));
}
.dnb-space__right--xx-large.dnb-space__right--x-large {
  margin-right: calc(var(--spacing-xx-large) + var(--spacing-x-large));
}
.dnb-space__right--xx-large-x2 {
  margin-right: calc(var(--spacing-x-large) + var(--spacing-large) + var(--spacing-large));
}
.dnb-space__right--xx-large-x2.dnb-space__right--x-small {
  margin-right: calc(var(--spacing-xx-large) + var(--spacing-large) + var(--spacing-large));
}
.dnb-space__right--xx-large-x2.dnb-space__right--small {
  margin-right: calc(var(--spacing-x-large) + var(--spacing-x-large) + var(--spacing-large));
}
.dnb-space__right--xx-large-x2.dnb-space__right--medium {
  margin-right: calc(var(--spacing-xx-large) + var(--spacing-x-large) + var(--spacing-large));
}
.dnb-space__right--xx-large-x2.dnb-space__right--large {
  margin-right: calc(var(--spacing-x-large) + var(--spacing-x-large) + var(--spacing-x-large));
}
.dnb-space__right--xx-large-x2.dnb-space__right--large.dnb-space__right--x-small {
  margin-right: calc(var(--spacing-xx-large) + var(--spacing-x-large) + var(--spacing-x-large));
}
.dnb-space__right--xx-large-x2.dnb-space__right--x-large {
  margin-right: calc(var(--spacing-xx-large) + var(--spacing-xx-large) + var(--spacing-x-large));
}
.dnb-space__bottom--zero {
  margin-bottom: 0;
}
.dnb-space__bottom--xx-small {
  margin-bottom: var(--spacing-xx-small);
}
.dnb-space__bottom--x-small {
  margin-bottom: var(--spacing-x-small);
}
.dnb-space__bottom--small {
  margin-bottom: var(--spacing-small);
}
.dnb-space__bottom--medium {
  margin-bottom: var(--spacing-medium);
}
.dnb-space__bottom--large {
  margin-bottom: var(--spacing-large);
}
.dnb-space__bottom--large.dnb-space__bottom--x-small {
  margin-bottom: calc(var(--spacing-large) + var(--spacing-x-small));
}
.dnb-space__bottom--x-large {
  margin-bottom: var(--spacing-x-large);
}
.dnb-space__bottom--xx-large {
  margin-bottom: var(--spacing-xx-large);
}
.dnb-space__bottom--xx-large.dnb-space__bottom--x-small {
  margin-bottom: calc(var(--spacing-large) + var(--spacing-large));
}
.dnb-space__bottom--xx-large.dnb-space__bottom--small {
  margin-bottom: calc(var(--spacing-x-large) + var(--spacing-medium));
}
.dnb-space__bottom--xx-large.dnb-space__bottom--medium {
  margin-bottom: calc(var(--spacing-x-large) + var(--spacing-large));
}
.dnb-space__bottom--xx-large.dnb-space__bottom--large {
  margin-bottom: calc(var(--spacing-x-large) + var(--spacing-large) + var(--spacing-x-small));
}
.dnb-space__bottom--xx-large.dnb-space__bottom--large.dnb-space__bottom--x-small {
  margin-bottom: calc(var(--spacing-x-large) + var(--spacing-large) + var(--spacing-small));
}
.dnb-space__bottom--xx-large.dnb-space__bottom--x-large {
  margin-bottom: calc(var(--spacing-xx-large) + var(--spacing-x-large));
}
.dnb-space__bottom--xx-large-x2 {
  margin-bottom: calc(var(--spacing-x-large) + var(--spacing-large) + var(--spacing-large));
}
.dnb-space__bottom--xx-large-x2.dnb-space__bottom--x-small {
  margin-bottom: calc(var(--spacing-xx-large) + var(--spacing-large) + var(--spacing-large));
}
.dnb-space__bottom--xx-large-x2.dnb-space__bottom--small {
  margin-bottom: calc(var(--spacing-x-large) + var(--spacing-x-large) + var(--spacing-large));
}
.dnb-space__bottom--xx-large-x2.dnb-space__bottom--medium {
  margin-bottom: calc(var(--spacing-xx-large) + var(--spacing-x-large) + var(--spacing-large));
}
.dnb-space__bottom--xx-large-x2.dnb-space__bottom--large {
  margin-bottom: calc(var(--spacing-x-large) + var(--spacing-x-large) + var(--spacing-x-large));
}
.dnb-space__bottom--xx-large-x2.dnb-space__bottom--large.dnb-space__bottom--x-small {
  margin-bottom: calc(var(--spacing-xx-large) + var(--spacing-x-large) + var(--spacing-x-large));
}
.dnb-space__bottom--xx-large-x2.dnb-space__bottom--x-large {
  margin-bottom: calc(var(--spacing-xx-large) + var(--spacing-xx-large) + var(--spacing-x-large));
}
.dnb-space__left--zero {
  margin-left: 0;
}
.dnb-space__left--xx-small {
  margin-left: var(--spacing-xx-small);
}
.dnb-space__left--x-small {
  margin-left: var(--spacing-x-small);
}
.dnb-space__left--small {
  margin-left: var(--spacing-small);
}
.dnb-space__left--medium {
  margin-left: var(--spacing-medium);
}
.dnb-space__left--large {
  margin-left: var(--spacing-large);
}
.dnb-space__left--large.dnb-space__left--x-small {
  margin-left: calc(var(--spacing-large) + var(--spacing-x-small));
}
.dnb-space__left--x-large {
  margin-left: var(--spacing-x-large);
}
.dnb-space__left--xx-large {
  margin-left: var(--spacing-xx-large);
}
.dnb-space__left--xx-large.dnb-space__left--x-small {
  margin-left: calc(var(--spacing-large) + var(--spacing-large));
}
.dnb-space__left--xx-large.dnb-space__left--small {
  margin-left: calc(var(--spacing-x-large) + var(--spacing-medium));
}
.dnb-space__left--xx-large.dnb-space__left--medium {
  margin-left: calc(var(--spacing-x-large) + var(--spacing-large));
}
.dnb-space__left--xx-large.dnb-space__left--large {
  margin-left: calc(var(--spacing-x-large) + var(--spacing-large) + var(--spacing-x-small));
}
.dnb-space__left--xx-large.dnb-space__left--large.dnb-space__left--x-small {
  margin-left: calc(var(--spacing-x-large) + var(--spacing-large) + var(--spacing-small));
}
.dnb-space__left--xx-large.dnb-space__left--x-large {
  margin-left: calc(var(--spacing-xx-large) + var(--spacing-x-large));
}
.dnb-space__left--xx-large-x2 {
  margin-left: calc(var(--spacing-x-large) + var(--spacing-large) + var(--spacing-large));
}
.dnb-space__left--xx-large-x2.dnb-space__left--x-small {
  margin-left: calc(var(--spacing-xx-large) + var(--spacing-large) + var(--spacing-large));
}
.dnb-space__left--xx-large-x2.dnb-space__left--small {
  margin-left: calc(var(--spacing-x-large) + var(--spacing-x-large) + var(--spacing-large));
}
.dnb-space__left--xx-large-x2.dnb-space__left--medium {
  margin-left: calc(var(--spacing-xx-large) + var(--spacing-x-large) + var(--spacing-large));
}
.dnb-space__left--xx-large-x2.dnb-space__left--large {
  margin-left: calc(var(--spacing-x-large) + var(--spacing-x-large) + var(--spacing-x-large));
}
.dnb-space__left--xx-large-x2.dnb-space__left--large.dnb-space__left--x-small {
  margin-left: calc(var(--spacing-xx-large) + var(--spacing-x-large) + var(--spacing-x-large));
}
.dnb-space__left--xx-large-x2.dnb-space__left--x-large {
  margin-left: calc(var(--spacing-xx-large) + var(--spacing-xx-large) + var(--spacing-x-large));
}
.dnb-space__reset:not([class*=dnb-space__top]) {
  margin-top: 0;
}
.dnb-space__reset:not([class*=dnb-space__bottom]) {
  margin-bottom: 0;
}
.dnb-space__reset:not([class*=dnb-space__left]) {
  margin-left: 0;
}
.dnb-space__reset:not([class*=dnb-space__right]) {
  margin-right: 0;
}
.dnb-space[style*="--space-t-"] {
  padding-top: var(--padding-top, 0);
}
.dnb-space[style*="--space-r-"] {
  padding-right: var(--padding-right, 0);
}
.dnb-space[style*="--space-b-"] {
  padding-bottom: var(--padding-bottom, 0);
}
.dnb-space[style*="--space-l-"] {
  padding-left: var(--padding-left, 0);
}
@media screen and (max-width: 60em) {
  .dnb-space {
    --padding-right: var(--space-r-s);
    --padding-left: var(--space-l-s);
    --padding-top: var(--space-t-s);
    --padding-bottom: var(--space-b-s);
  }
}
@media screen and (max-width: 60em) and (min-width: 40.00625em) {
  .dnb-space {
    --padding-right: var(--space-r-m);
    --padding-left: var(--space-l-m);
    --padding-top: var(--space-t-m);
    --padding-bottom: var(--space-b-m);
  }
}
@media screen and (min-width: 60.00625em) {
  .dnb-space {
    --padding-right: var(--space-r-l);
    --padding-left: var(--space-l-l);
    --padding-top: var(--space-t-l);
    --padding-bottom: var(--space-b-l);
  }
}

span.dnb-space--no-collapse > span {
  display: block;
}

/* stylelint-enable */"
`;

exports[`Space scss have to match default theme snapshot 1`] = `
"/*
 * Np theme is provided
 *
 */"
`;
